internal/service/url_service_test.go

```diff
@@ -25,7 +34,63 @@ func (m *MockURLRepository) GetStats(shortURL string) (models.StatsResponse, err
 	return args.Get(0).(models.StatsResponse), args.Error(1)
 }
 
-// TODO TestCreateShortURL and TestGetOriginalURL
+func TestSaveURL(t *testing.T) {
+	mockRepo := new(MockURLRepository)
+	service := NewURLService(mockRepo)
+
+	OriginalURL := "http://testurl.com"
+	castomalias := "http://myurl123"
+	shortURL := "http://abc123"
+
+	shortrequesr := models.ShortenRequest{
+		URL:         OriginalURL,
+		CustomAlias: &castomalias,
+	}
+
+	testURL := models.URL{
+		ID:          "testone",
+		OriginalURL: shortrequesr.URL,
+		ShortURL:    shortURL,
+		CustomAlias: nil,
+		CreatedAt:   time.Now(),
+		ExpiredAt:   nil,
+	}
+	// TODO FIX THIS....
+	if shortrequesr.CustomAlias != nil && *shortrequesr.CustomAlias != "" {
```

Comment:
it is not necessary to write if/else when you want to test several scenarios, it is better to write separate tests, if/else takes place when we have table tests with a huge number of different variations of the input data

Reply:
fix all TODO

Reply:
fixed